libraries/AP_HAL_ESP32/Util.cpp

```diff
@@ -243,8 +243,7 @@ bool Util::get_system_id(char buf[50])
 
 bool Util::get_system_id_unformatted(uint8_t buf[], uint8_t &len)
 {
-    len = MIN(12, len);
-
+    len = MIN(6, len);
```

Comment:
This would be better done as `len = MIN(len, ARRAY_SIZE(base_mac_addr)`, and moved down to where we are doing the mempy.

Not a blocker, this is good-enough as-is IMO